exp/addons/api/v1beta1/clusterresourcesetbinding_webhook.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"sigs.k8s.io/cluster-api/feature"
+)
+
+func (c *ClusterResourceSetBinding) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(c).
+		Complete()
+}
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-addons-cluster-x-k8s-io-v1beta1-clusterresourcesetbinding,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=addons.cluster.x-k8s.io,resources=clusterresourcesetbindings,versions=v1beta1,name=validation.clusterresourcesetbinding.addons.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
+
+var _ webhook.Validator = &ClusterResourceSetBinding{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
+func (c *ClusterResourceSetBinding) ValidateCreate() error {
+	return c.validate(nil)
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
+func (c *ClusterResourceSetBinding) ValidateUpdate(old runtime.Object) error {
+	oldBinding, ok := old.(*ClusterResourceSetBinding)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a ClusterResourceSetBinding but got a %T", old))
+	}
+	return c.validate(oldBinding)
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
+func (c *ClusterResourceSetBinding) ValidateDelete() error {
+	return nil
+}
+
+func (c *ClusterResourceSetBinding) validate(old *ClusterResourceSetBinding) error {
+	// NOTE: ClusterResourceSet is behind ClusterResourceSet feature gate flag; the web hook
+	// must prevent creating new objects in case the feature flag is disabled.
+	if !feature.Gates.Enabled(feature.ClusterResourceSet) {
+		return field.Forbidden(
+			field.NewPath("spec"),
+			"can be set only if the ClusterResourceSet feature flag is enabled",
+		)
+	}
+	var allErrs field.ErrorList
+	if old != nil && old.Spec.ClusterName != nil && c.Spec.ClusterName != nil && *old.Spec.ClusterName != *c.Spec.ClusterName {
```

Comment:
Looks like we allow an existing ClusterName field to be set to empty. Is this intentional?

Reply:
> Looks like we allow an existing ClusterName field to be set to empty. 


I don't think this will happen. because of `*old.Spec.ClusterName != *c.Spec.ClusterName`

Reply:
Sorry I wasn't precise enough. What I meant is: If c.Spec.ClusterName is nil, this statement won't return an error.

(Let's wait for the result of the pointer discussion, this might change this code anyway)